src/enzyme_ad/jax/Utils.h

```diff
@@ -454,9 +456,13 @@ template <typename Child> class GuaranteedResultAnalysisBase {
       case State::PENDING: {
         assert(localtodo.size());
         assert(seen.find(cur) == seen.end());
-        SmallPtrSet<Operation *, 2> set(localtodo.begin(), localtodo.end());
-        for (auto v : set) {
+        llvm::SmallSetVector<Operation *, 2> set(localtodo.begin(),
```

Comment:
so this I don't think needs to be a setvector, the change on 458 of set -> localtodo should keep fix the nondeterminsim on own?